lib/webpage_cache_service/cache.rb

```diff
@@ -79,42 +83,67 @@ def client
     def get uri:
       response = client.get uri
 
-      content_type = response.content_type.mime_type
+      # Without manually managing the temp file, there isn't an easy way to
+      # breakup the following logic but I did my best and fuck you too rubocop
+      attachments = temp_file_for uri: uri do |temp_file, uri_filename|
+        write_body response: response, to: temp_file
 
-      temp_key = Digest::SHA256.hexdigest uri.to_s
-      attachment = Tempfile.create temp_key do |temp_file|
-        temp_file.binmode
+        offline_cache.assets.attach(
+          io: temp_file,
+          filename: uri_filename,
+          content_type: get_content_type(response: response, io: temp_file),
+          metadata: build_metadata(response: response)
+        )
+      end
+
+      [ response, attachments.first ]
+    end
 
-        response.body.each do |partial|
-          temp_file.write partial
-        end
+    def temp_file_for uri:
+      uri_filename = Digest::SHA256.hexdigest uri.to_s
```

Comment:
if its a tempfile, why does the name matter

Reply:
Apparently, in [2.4](https://blog.bigbinary.com/2017/04/04/ruby-2-4-has-default-basename-for-tempfile-create.html) it doesn't but it was required in previous versions of `Tempfile`. TIL